### ISSUE TITLE
ansible-lint fails to identify basic syntax errors in conditions 

### ISSUE THREAD

**Issue by ssbarnea** [ansible/ansible-lint#353]
Today I discovered that ansible-lint missed to identify a very simple typo in when condition: 
```
    when: item i success
```
I think that this kind of error should be found during linting as it could save a good number of hours of testing.

**Comment by willthames** [comment]
I suspect solving that one would involve some horrendous descent into jinja parsing (bearing in mind that it's incredibly difficult to parse that when condition without knowing that `item` is a variable, `success` is a jinja test and `i` is neither of those but a typo for a jinja keyword (I think `is` is jinja, but I'm not 100% sure)

If someone can solve this one, sure, I'd take a PR for it, as long as it doesn't cause any additional false positives.
